### ISSUE TITLE
[duckdb] Found unparsable section "SET VARIABLE"

### ISSUE THREAD

**Issue by cuong0993** [sqlfluff/sqlfluff#7325]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff can not parse "SET VARIABLE" in duckdb.

### Expected Behaviour

sqlfluff should parse successfully.

### Observed Behaviour

L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'SET                                                                                        
                       | VARIABLE my_var = 30;\nSELECT 20 + ge...'

### How to reproduce

Run sqlfluff lint

```sql
SET VARIABLE my_var = 30;
SELECT 20 + getvariable('my_var') AS total;
```

### Dialect

duckdb

### Version

3.5.0

### Configuration

[sqlfluff]
dialect = duckdb

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
